query-node/schema.graphql

```diff
@@ -12,32 +12,127 @@ type Block @entity {
   network: Network!
 }
 
+enum MembershipEntryMethod {
+  PAID
+  SCREENING
+  GENESIS
+}
+
 "Stored information about a registered user"
-type Member @entity {
+type Membership @entity {
   "MemberId: runtime identifier for a user"
   id: ID!
 
   "The unique handle chosen by member"
-  handle: String @unique @fulltext(query: "membersByHandle")
+  handle: String! @unique @fulltext(query: "membersByHandle")
 
   "A Url to member's Avatar image"
-  avatarUri: String
+  avatarUri: String!
 
   "Short text chosen by member to share information about themselves"
-  about: String
-
-  "Blocknumber when member was registered"
-  registeredAtBlock: Int!
+  about: String!
 
   "Member's controller account id"
-  controllerAccount: Bytes!
+  controllerAccount: String!
 
   "Member's root account id"
-  rootAccount: Bytes!
+  rootAccount: String!
 
-  happenedIn: Block!
+  "Blocknumber when member was registered"
+  registeredAtBlock: Block!
+
+  "Timestamp when member was registered"
+  registeredAtTime: BigInt!
+
+  "How the member was registered"
+  entry: MembershipEntryMethod!
+
+  "The type of subscription the member has purchased if any."
+  subscription: BigInt
+}
+
+# TODO: add description to fields
+# TODO: revise property requirement for the new and existing fields derived from content-metadata-protobuf
+
+type Account @entity {
```

Comment:
This type isn't used anywhere. What is it for?

Reply:
I forgot to remove this one. Initially, I thought an entity for an account would be handy there, but since it has no properties, it is redundant. Done.